### PR TITLE
Refactor User Groups with split view approach in to feature libs #8190

### LINKED ISSUE #8190: Refactor User Groups with split view approach in to feature libs
- [x] Listing
- [x] Details
- [x] Create
- [x] Edit
- [x] Users
- [x] Assign Users
- [x] Purchase limtis
- [x] Assign Purchase limtis

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #8190

### CODE REVIEW

**Review by Matejk00 — APPROVED** [review]

File: feature-libs/my-account/organization/src/components/user-group/form/user-group-form.module.ts [hunk 1]
```diff
@@ -17,12 +21,11 @@ import { FormErrorsModule } from '../../../../shared/components/form/form-errors
     UrlModule,
     I18nModule,
     ReactiveFormsModule,
-    DatePickerModule,
     FormErrorsModule,
   ],
   declarations: [UserGroupFormComponent],
   exports: [UserGroupFormComponent],
-  providers: [OrgUnitService],
+  providers: [CurrencyService, OrgUnitService],
```
Comment (Platonn) [inline feature-libs/my-account/organization/src/components/user-group/form/user-group-form.module.ts:28]: Do we really need to provide `CurrencyService` here?

File: feature-libs/my-account/organization/src/components/user-group/create/user-group-create.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    protected userGroupFormService: UserGroupFormService,
+    protected routingService: RoutingService
+  ) {}
+
+  save(form: FormGroup): void {
+    if (form.invalid) {
+      form.markAllAsTouched();
+    } else {
+      form.disable();
+      console.log(form.value);
```
Comment (Platonn) [inline feature-libs/my-account/organization/src/components/user-group/create/user-group-create.component.ts:40]: 👀

File: feature-libs/my-account/organization/src/components/shared/organization.model.ts [hunk 3]
```diff
@@ -1,5 +1,10 @@
 export enum OrganizationTableType {
   COST_CENTER = 'costCenter',
   COST_CENTER_BUDGETS = 'costCenterBudgets',
-  COST_CENTER_ASSIGN_BUDGET = 'costCenterAssignBudget',
+  COST_CENTER_ASSIGN_BUDGET = 'costCenterAssignBudgets',
+  USER_GROUP = 'userGroup',
+  USER_GROUP_USERS = 'userGroupUsers',
+  USER_GROUP_ASSIGN_USER = 'userGroupAssignUsers',
```
Comment (Platonn) [inline feature-libs/my-account/organization/src/components/shared/organization.model.ts:7]: For consistency, can we align the keys with the values (i,e. key ....USER vs value ...Users).
Same for COST_CENTER_ASSIGN_BUDGET

File: feature-libs/my-account/organization/src/components/user-group/user-group.config.ts [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+};
+
+export function userGroupTableConfigFactory(): TableConfig {
+  return userGroupTableConfig;
+}
+
+export const userGroupTableConfig: TableConfig = {
+  table: {
+    [OrganizationTableType.USER_GROUP]: [
+      // TODO: consider cascading from smallest size
```
Comment (Platonn) [inline feature-libs/my-account/organization/src/components/user-group/user-group.config.ts:117]: This comment is not needed for cost center anymore and shouldn't be multiplicated for other features.

File: feature-libs/my-account/organization/src/components/user-group/user-group.config.ts [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+      {
+        headers: [{ key: 'name' }],
+        pagination: {
+          sort: 'byName',
+          // pageSize: 2,
```
Comment (Platonn) [inline feature-libs/my-account/organization/src/components/user-group/user-group.config.ts:122]: 👀

File: feature-libs/my-account/organization/src/components/user-group/form/user-group-form.module.ts (hunk 1 above)
Comment (Pucek9) [reply feature-libs/my-account/organization/src/components/user-group/form/user-group-form.module.ts:28]: no

File: feature-libs/my-account/organization/src/components/user-group/list/user-group-list.component.html [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+<ng-template cxOutletRef="table.userGroup.data.name" let-model>
+  <div routerLinkActive="active">
+    <a [routerLink]="{ cxRoute: 'userGroupDetails', params: model } | cxUrl">
+      <span class="text">{{ model.name }}</span>
+      <cx-icon type="CARET_RIGHT"></cx-icon>
+    </a>
+  </div>
+</ng-template>
+
+<!-- TODO: refactor outlet concatenation-->
```
Comment (Platonn) [inline feature-libs/my-account/organization/src/components/user-group/list/user-group-list.component.html:49]: I believe we shouldn't multiplicate this comment too.

**Pull request merged**